src/Modules/Manage/Dnn.PersonaBar.Users/Components/UsersController.cs

```diff
@@ -488,24 +490,39 @@ private static bool IsAdmin(UserInfo user, PortalSettings portalSettings)
             return user.IsSuperUser || user.IsInRole(portalSettings.AdministratorRoleName);
         }
 
-        private static IEnumerable<UserBasicDto> GetUsers(GetUsersContract usersContract,
+        private IEnumerable<UserBasicDto> GetUsers(GetUsersContract usersContract,
             bool? includeAuthorized, bool? includeDeleted, bool? includeSuperUsers, out int totalRecords)
         {
+
+            var parsedSearchText = string.IsNullOrEmpty(usersContract.SearchText) ? "" : SearchTextFilter.Apply(usersContract.SearchText.Trim());
+
             List<UserBasicDto2> records = CBO.FillCollection<UserBasicDto2>(
-                DataProvider.Instance().ExecuteReader(
+                CallGetUsersBySearchTerm(
+                    usersContract,
+                    includeAuthorized, 
+                    includeDeleted, 
+                    includeSuperUsers));
+
+            totalRecords = records.Count == 0 ? 0 : records[0].TotalCount;
+            return records;
+        }
+
+        protected virtual IDataReader CallGetUsersBySearchTerm(GetUsersContract usersContract,
+            bool? includeAuthorized, bool? includeDeleted, bool? includeSuperUsers)
+        {
+            var parsedSearchText = string.IsNullOrEmpty(usersContract.SearchText) ? "" : SearchTextFilter.Apply(usersContract.SearchText.Trim());
```

Comment:
To follow up on the other comment, I was expecting that the filtered text would become a parameter of `CallGetUsersBySearchTerm`. This way you only wrap the `DataProvider.Instance().ExecuteReader()` call and nothing else.

```c#
var parsedSearchText = SearchTextFilter.Apply(usersContract.SearchText.Trim());

CallGetUsersBySearchTerm(
    usersContract,
    parsedSearchText
    includeAuthorized, 
    includeDeleted, 
    includeSuperUsers));
```

This way you will be able to verify that the controller passes the expected search text to the stored procedure.

Reply:
Filtered test is an attribute of UsersContract. Don't need to make it an argument, IMO.

Reply:
Inside UsersController.CallGetUsersBySearchTerm() we actually call the DataProvider. Nothing else happens in the middle, so I think it's enough to test the value of *parsedSearchText* that way.
We care of the resulting string actually.
``` c#
protected virtual IDataReader CallGetUsersBySearchTerm(GetUsersContract usersContract,
            bool? includeAuthorized, bool? includeDeleted, bool? includeSuperUsers)
        {
            var parsedSearchText = string.IsNullOrEmpty(usersContract.SearchText) ? "" : SearchTextFilter.CleanWildcards(usersContract.SearchText.Trim());

            return DataProvider.Instance().ExecuteReader(
                    "Personabar_GetUsersBySearchTerm",
                    usersContract.PortalId,
                    string.IsNullOrEmpty(usersContract.SortColumn) ? "Joined" : usersContract.SortColumn,
                    usersContract.SortAscending,
                    usersContract.PageIndex,
                    usersContract.PageSize,
                    parsedSearchText,
                    includeAuthorized,
                    includeDeleted,
                    includeSuperUsers);
        }
```